GPU/GPUTracking/dEdx/TPCdEdxCalibrationSplines.h

```diff
@@ -0,0 +1,177 @@
+namespace gpu
+{
+
+///
+/// The dEdxCalibrationSplines class represents the calibration of the dEdx of mostly geometrical effects
+///
+
+class TPCdEdxCalibrationSplines : public FlatObject
+{
+ public:
+  typedef Spline2D<float, 1, 1> SplineType;
+
+  /// _____________  Constructors / destructors __________________________
+
+#if !defined(GPUCA_GPUCODE)
+  /// Default constructor
+  TPCdEdxCalibrationSplines();
+
+  /// Copy constructor
+  TPCdEdxCalibrationSplines(const TPCdEdxCalibrationSplines&);
+
+  /// Assignment operator
+  TPCdEdxCalibrationSplines& operator=(const TPCdEdxCalibrationSplines&);
+
+  void recreate(int nKnotsU1[], int nKnotsU2[]);
+#else
+  /// Disable constructors for the GPU implementation
+
+  TPCdEdxCalibrationSplines() CON_DELETE;
+  TPCdEdxCalibrationSplines(const TPCdEdxCalibrationSplines&) CON_DELETE;
+  TPCdEdxCalibrationSplines& operator=(const TPCdEdxCalibrationSplines&) CON_DELETE;
+#endif
+
+  /// Destructor
+  ~TPCdEdxCalibrationSplines() CON_DEFAULT;
+
+  /// _____________  FlatObject functionality, see FlatObject class for description  ____________
+
+  using FlatObject::getBufferAlignmentBytes;
+  using FlatObject::getClassAlignmentBytes;
+
+#if !defined(GPUCA_GPUCODE)
+  void cloneFromObject(const TPCdEdxCalibrationSplines& obj, char* newFlatBufferPtr);
+  void moveBufferTo(char* newBufferPtr);
+#endif
+
+  using FlatObject::releaseInternalBuffer;
+
+  void destroy();
+  void setActualBufferAddress(char* actualFlatBufferPtr);
+  void setFutureBufferAddress(char* futureFlatBufferPtr);
+
+  /// ______________
+
+  /// Gives pointer to a spline
+  GPUd() const SplineType& getSpline(int chargeType, int region) const;
+
+#if !defined(GPUCA_GPUCODE) && !defined(GPUCA_STANDALONE)
+  /// sets the splines from an input file
+  void setSplinesFromFile(TFile& inpf);
+#endif
+
+  /// returns the number of splines stored in the calibration object
+  GPUd() unsigned int getFSplines() const
+  {
+    return mFSplines;
+  };
+
+  GPUd() float interpolateqMax(const int splineInd, const float angleZ, const float z) const
+  {
+    return mCalibSplinesqMax[splineInd].interpolate(angleZ, z);
+  };
+
+  GPUd() float interpolateqTot(const int splineInd, const float angleZ, const float z) const
+  {
+    return mCalibSplinesqTot[splineInd].interpolate(angleZ, z);
+  };
+
+  GPUd() SplineType& getSplineqMax(const int splineInd)
+  {
+    return mCalibSplinesqMax[splineInd];
+  };
+
+  GPUd() SplineType& getSplineqTot(const int splineInd)
+  {
+    return mCalibSplinesqTot[splineInd];
+  };
+
+    /// _______________  IO   ________________________
+#if !defined(GPUCA_GPUCODE) && !defined(GPUCA_STANDALONE)
+  /// write a class object to the file
+  int writeToFile(TFile& outf, const char* name);
+
+  /// read a class object from the file
+  static TPCdEdxCalibrationSplines* readFromFile(TFile& inpf, const char* name);
+#endif
+
+ private:
+  constexpr static unsigned int mFSplines = 10; ///< number of splines stored for each type
```

Comment:
I think constexpr should not start with `m`, but either `s` or simply with a capital letter. See:
https://rawgit.com/AliceO2Group/CodingGuidelines/master/naming_formatting.html?showone=Variable_Names#Variable_Names